includes/models/model.llms.student.php

```diff
@@ -1433,38 +1434,55 @@ public function mark_incomplete( $object_id, $object_type, $trigger = 'unspecifi
 	 */
 	private function remove_membership_level( $membership_id, $status = 'expired', $delete = false ) {
 
-		// Remove the user from the membership level.
+		global $wpdb;
+
+		// Remove the user from the given membership level.
 		$membership_levels = $this->get_membership_levels();
-		$key               = array_search( $membership_id, $membership_levels );
+		$key               = array_search( $membership_id, $membership_levels, true );
 		if ( false !== $key ) {
 			unset( $membership_levels[ $key ] );
 		}
 		update_user_meta( $this->get_id(), '_llms_restricted_levels', $membership_levels );
 
-		global $wpdb;
-		// Locate all enrollments triggered by this membership level.
-		$q = $wpdb->get_results(
+		// Fetch all of this user's enrolled course IDs triggered by any membership.
+		$course_ids_enrolled_by_any_membership = $wpdb->get_col(
 			$wpdb->prepare(
-				"SELECT post_id FROM {$wpdb->prefix}lifterlms_user_postmeta WHERE user_id = %d AND meta_key = '_enrollment_trigger' AND meta_value = %s",
-				array( $this->get_id(), 'membership_' . $membership_id )
-			),
-			'OBJECT_K'
+				"SELECT post_id FROM {$wpdb->prefix}lifterlms_user_postmeta " .
+				"WHERE user_id = %d AND meta_key = '_enrollment_trigger' AND meta_value LIKE %s",
+				array( $this->get_id(), 'membership_%' )
+			)
 		); // db call ok; no-cache ok.
+		$course_ids_enrolled_by_any_membership = array_map( 'intval', $course_ids_enrolled_by_any_membership ); // Cast strings to integers.
 
-		$courses = array_keys( $q );
+		// Get the unique auto-enrollment course IDs, with any post status, of other membership levels.
+		$other_memberships_auto_enroll_course_ids = array();
+		foreach ( $membership_levels as $membership_level ) {
+			$membership_auto_enroll_course_ids         = ( new LLMS_Membership( $membership_level ) )->get( 'auto_enroll' );
+			$other_memberships_auto_enroll_course_ids += array_flip( $membership_auto_enroll_course_ids );
+		}
 
-		if ( $courses ) {
+		// Get the auto-enrollment course IDs, with any post status, of the given membership.
+		$given_membership_auto_enroll_course_ids = ( new LLMS_Membership( $membership_id ) )->get( 'auto_enroll' );
 
-			// Loop through all the courses and update the enrollment status.
-			foreach ( $courses  as $course_id ) {
-				if ( ! $delete ) {
-					$this->unenroll( $course_id, 'membership_' . $membership_id, $status );
-				} else {
-					$this->delete_enrollment( $course_id, 'membership_' . $membership_id );
-				}
+		// Loop through all the auto-enrolled courses and update the enrollment status.
+		foreach ( $given_membership_auto_enroll_course_ids as $given_membership_auto_enroll_course_id ) {
+
+			// Do not change course enrollment if the user does not have one. It may have been deleted manually.
```

Comment:
I think I was trying to not waste time changing an enrollment if the user isn't enrolled in the course. It may improve performance a very small amount, but probably isn't needed. If you agree, I will take the check out.

Reply:
Right, we don't want the student to be unenrolled from a course in that use case.

Your first comment is correct and my response was incorrect.

The check is really doing this:
`// Do not change course enrollment if not triggered by a membership.`